### PR TITLE
docs, quiltctl: Warn that Vagrant isn't ready

### PR DESCRIPTION
**Pull request by sirspinach**

### CODE REVIEW

File: quiltctl/command/run.go [hunk 1]
```diff
@@ -96,5 +97,14 @@ func (rCmd *Run) Run() int {
 	}
 
 	fmt.Println("Successfully started run.")
+
+	//XXX: Remove warning when Vagrant provider is done.
+	for _, machine := range compiled.Machines {
```
Comment (kklin) [inline quiltctl/command/run.go]: I have a slight preference for returning an error in the QRI (`api/server`) I think. Could also be easily unit tested there.

**Comment by sirspinach** [comment]
Ready for further review.

File: api/server/server.go [hunk 2]
```diff
@@ -125,5 +125,16 @@ func (s server) Deploy(cts context.Context, deployReq *pb.DeployRequest) (
 		return &pb.DeployReply{}, err
 	}
 
+	// XXX: Remove this warning when Vagrant provider is done.
+	// Allow UpdatePolicy on stitches with Vagrant machines, but send an error back
+	// to the client as a "warning" to users.
+	for _, machine := range stitch.Machines {
+		if machine.Provider == db.Vagrant {
+			err = errors.New("Vagrant provider is in development." +
```
Comment (kklin) [inline api/server/server.go]: My personal rule of thumb is that when I feel the need to write a comment, it might be a sign that the code could be clearer. In this case, I think we could rephrase this error to be more clear that the stitch will still be deployed.

File: api/server/server_test.go [hunk 3]
```diff
@@ -103,13 +76,37 @@ func TestDeploy(t *testing.T) {
...
+	}]}`
+	vagrantErrMsg := "Vagrant provider is in development. Please choose the" +
+		" Amazon or Google provider."
+	testDeploy(vagrantDeployment, 2, vagrantErrMsg, t)
+}
 
-	if err != nil {
-		t.Errorf("Unexpected error when deploying stitch: %s\n", err.Error())
-		return
+func testDeploy(deployment string, expNumMachines int, expErrMsg string,
```
Comment (kklin) [inline api/server/server_test.go]: Everywhere else in the codebase has `*testing.T` as the first arg, may as well be consistent.

File: api/server/server_test.go [hunk 4]
```diff
@@ -103,13 +76,37 @@ func TestDeploy(t *testing.T) {
...
+	conn := db.New()
+	s := server{dbConn: conn}
+
+	_, err := s.Deploy(context.Background(),
+		&pb.DeployRequest{Deployment: deployment})
+
+	if expErrMsg == "" {
+		assert.Nil(t, err, "didn't expect a deployment error")
+	} else {
+		if assert.NotNil(t, err, "expected a deployment error") {
```
Comment (kklin) [inline api/server/server_test.go]: There's an `assert.EqualError`

File: api/server/server_test.go [hunk 5]
```diff
@@ -118,10 +115,7 @@ func TestDeploy(t *testing.T) {
 		return nil
 	})
 
-	if len(machines) != 2 {
-		t.Errorf("Two machines should have been created by the deployment, "+
-			"but we found: %v\n", machines)
-	}
+	assert.Equal(t, expNumMachines, len(machines))
```
Comment (kklin) [inline api/server/server_test.go]: `assert.Len`

File: docs/Vagrant.md [hunk 6]
```diff
@@ -1,4 +1,7 @@
 # Vagrant
+NOTE: Vagrant backend is not yet finished. If you are looking to try Quilt out,
```
Comment (kklin) [inline docs/Vagrant.md]: "The Vagrant"

File: api/server/server.go (hunk 2 above)
Comment (kklin) [inline api/server/server.go]: Also, "The Vagrant provider"

**Comment by sirspinach** [comment]
ready for more review!

File: api/server/server.go [hunk 7]
```diff
@@ -125,5 +125,16 @@ func (s server) Deploy(cts context.Context, deployReq *pb.DeployRequest) (
 		return &pb.DeployReply{}, err
 	}
 
+	// XXX: Remove this error when the Vagrant provider is done.
+	for _, machine := range stitch.Machines {
+		if machine.Provider == db.Vagrant {
+			err = errors.New("The Vagrant provider is in development." +
+				" The stitch will continue to run, but" +
+				" probably won't work correctly." +
+				" Please choose the Amazon or Google provider.")
```
Comment (ejj) [inline api/server/server.go]: I'd delete this last line, otherwise we're going to have to remember to update this error every time we add a cloud provider.

File: docs/Vagrant.md [hunk 8]
```diff
@@ -1,4 +1,7 @@
 # Vagrant
+NOTE: The Vagrant backend is not yet finished. If you are looking to try Quilt out,
+please use the Amazon EC2 or Google Compute backend, as demonstrated in [Getting
```
Comment (ejj) [inline docs/Vagrant.md]: Again, here I'd ditch the amazon or google reference and just say "a different provider"

File: quiltctl/command/run.go [hunk 9]
```diff
@@ -91,7 +91,7 @@ func (rCmd *Run) Run() int {
 
 	err = c.Deploy(compiled.String())
 	if err != nil {
-		log.WithError(err).Error("Unable to start run.")
+		log.WithError(err).Error("Error while starting run.")
```
Comment (ejj) [inline quiltctl/command/run.go]: This change doesn't really belong in this patch IMO.  Also it's not clear to me that it's better.

Comment (sirspinach) [inline quiltctl/command/run.go]: When we run a stitch with the Vagrant provider, we return an error which is logged at this line, but we don't abort the run. 

So "Unable to start run" is no longer accurate here.

**Comment by ejj** [comment]
@sirspinach, looked it over .. the patch seems basically ready, though there's a merge conflict now.  Rebase and resubmit the PR and I'll merge.

**Comment by sirspinach** [comment]
This series should be ready to merge now, @ejj
